sherpa/conftest.py

```diff
@@ -42,13 +42,18 @@
     has_xspec = False
 
 
-import sherpa.plot
-import sherpa.astro.plot
-from sherpa.plot import dummy_backend
+from sherpa.plot.backends import BaseBackend, IndepOnlyBackend
+from sherpa.plot import TemporaryPlottingBackend
+
+ALL_PLOT_BACKENDS = [BaseBackend(), IndepOnlyBackend()]
+PLOT_BACKENDS = [BaseBackend()]
+
 try:
-    from sherpa.plot import pylab_backend
+    from sherpa.plot.pylab_backend import PylabBackend
+    ALL_PLOT_BACKENDS.append(PylabBackend())
+    PLOT_BACKENDS.append(PylabBackend())
 except ImportError:
-    pylab_backend = None
+    pass
```

Comment:
So, is the idea that when you add a new backend you should add it to the `*_BACKENDS` arrays here? I know that I added `PLOT_BACKENDS`, not y, but just want to check.

Could we do something like (pseudo code, depends on exactly what `sherpa.plot.backends` contains)

```
from sherpa.plot import plot_backends

for k, v in plot_backends:
    try:
        backend = some-sort-of-import/creation based on v
    except ImportError:
        pass
    ALL_PLOT_BACKENDS.append(backend)
    PLOT_BACKENDS.append(backend)
``` 

which would make this "automatic" without the backend writer having to know about this? Perhaps it's too clever (but I do want to make it easy for a user to select a different backend, and we could use that code here).

Reply:
This would be great if we could auto-populate plot_backends - as I said above there is no simple solution for that. A custom metaclass on `BasicBackend` (and thus automatically on all derived and imported classes, even if written outside of sherpa) would work. entry points are designed for this, but require either two implementations, a new dependency, or Python >=3.10.

Without that, you just kick down the can to add the backend to `plot_backends` and the user needs to know about that... so I think I have a solution I'll implement (metaclass with registry) so you can stop commenting on this part now ;-)
Thanks for bringing it up though - it was one of the things I considered and discarded as "probably not worth the effort", but apparently both of us think that it might be useful, so I'll do it.